src/java.desktop/windows/classes/sun/awt/shell/Win32ShellFolder2.java

```diff
@@ -1214,15 +1217,15 @@ static Image getSystemIcon(SystemIcon iconType) {
      */
     static Image getShell32Icon(int iconID, int size) {
         long hIcon = getIconResource("shell32.dll", iconID, size, size);
-        if (hIcon != 0) {
-            Image icon = makeIcon(hIcon);
-            if (size != icon.getWidth(null)) {
-                icon = new MultiResolutionIconImage(size, icon);
-            }
-            disposeIcon(hIcon);
-            return icon;
+        if (hIcon == 0) {
+            return null;
         }
```

Comment:
```suggestion
        if (hIcon == 0) {
            return null;
        }

```
And I'd add a blank line here, too.

Reply:
Updated.